share/sysbuild/cmake/modules/sysbuild_extensions.cmake

```diff
@@ -199,7 +199,9 @@ endfunction()
 # build system
 #
 # APPLICATION: <name>:       Name of the application, name will also be used for build
-#                            folder of the application
+#                            folder of the application.
+#                            The following names are not allowed: "SB", "all".
```

Comment:
> if we want to list reserved names, then please note that `sysbuild` is also reserved.

I didn't know that. Building an image named `sysbuild` in sysbuild appears to work fine today. Should we reserve that name anyway?

> and technically any target name already created by the build system

True, but the point is to reserve names that don't conflict with the build system per se, but which can cause confusion. `SB` and `all` are only reserved so that if you set e.g., `SB_CONFIG=value`, then there's no question as to where the value should be applied.

Adding an image named after an existing target will already fail with either:
```
CMake Error at cmake/modules/sysbuild_extensions.cmake:240 (message):
  ExternalZephyrProject_Add(APPLICATION boards ...) already exists.
  Application names must be unique.
```
or (for targets that are added inside `ExternalZephyrProject_Add()`):
```
CMake Error at /usr/share/cmake-3.25/Modules/ExternalProject.cmake:3985 (add_custom_target):
  add_custom_target cannot create target "menuconfig" because another target
  with the same name already exists.  The existing target is a custom target
  created in source directory "/home/grsw/ncs/zephyr/share/sysbuild/images".
  See documentation for policy CMP0002 for more details.
```
These error messages aren't ideal, but that can be improved on a separate occasion.

Reply:
> Building an image named sysbuild in sysbuild appears to work fine today. 

how did you test that ?
Cause with sysbuild we reserve `<image>_menuconfig` for launching `menuconfig` for a given target, but for sysbuild itself we have already reserved `sysbuild_menuconfig`, meaning `sysbuild_menuconfig` becomes ambigous.
Does it refer to sysbuild's menuconfig or the image named sysbuild's menuconfig ?

Reply:
Ah, I forgot about `sysbuild_menuconfig`. That falls in line with the sort of ambiguity I wanted to prevent here, which means I could add `sysbuild` as a reserved name for target namespacing.

Normally, the two targets will conflict when `<image> == sysbuild`, but there is an exception - when `<image>` is the main app, then `<image>_menuconfig` is never created. That was the only scenario I tested earlier today :[

Relevant code:
https://github.com/zephyrproject-rtos/zephyr/blob/27d519b260b25826689a4e5283af8a7f188d121c/share/sysbuild/cmake/modules/sysbuild_extensions.cmake#L340-L344